File: ContactList/app/src/main/java/com/bergamin/contactlist/receiver/SMSReceiver.java
package com.bergamin.contactlist.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.media.MediaPlayer;
import android.telephony.SmsMessage;
import android.widget.Toast;

import com.bergamin.contactlist.R;
import com.bergamin.contactlist.dao.ContactDAO;

/**
 * Created by Guilherme on 20/09/2016.
 */
public class SMSReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
            Object[] pdus = (Object[]) intent.getSerializableExtra("pdus");
            byte[] pdu = (byte[]) pdus[0];
            String format = (String) intent.getSerializableExtra("format");
            SmsMessage sms = null;
            sms = SmsMessage.createFromPdu(pdu, format);
            ContactDAO dao = new ContactDAO(context);
            if (dao.isContact(sms.getDisplayOriginatingAddress())) {
                Toast.makeText(context, "You have a new SMS from a contact", Toast.LENGTH_SHORT).show();
                MediaPlayer mediaPlayer = MediaPlayer.create(context, R.raw.msg);
                mediaPlayer.start();
            }
            dao.close();
        }
    }
}

File: Kotlin/ContactList/app/src/main/java/com/bergamin/contactlist/receiver/SMSReceiver.kt
package com.bergamin.contactlist.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.telephony.SmsMessage
import android.widget.Toast
import com.bergamin.contactlist.R
import com.bergamin.contactlist.dao.ContactDAO

/**
 * Created by Guilherme Taffarel Bergamin on 30/11/2017.
 */
class SMSReceiver: BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent?) {
        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            val pdus = intent?.getSerializableExtra("pdus") as Array<*>
            val pdu = pdus[0] as ByteArray
            val format = intent.getSerializableExtra("format") as String
            val sms = SmsMessage.createFromPdu(pdu, format)
            val dao = ContactDAO(context!!)

            if(dao.isContact(sms.displayOriginatingAddress)){
                Toast.makeText(context,context.getString(R.string.new_sms_received),Toast.LENGTH_SHORT).show()
            }
        }
    }
}
